apps/web/modules/users/views/users-type-public-view.getServerSideProps.tsx

```diff
@@ -160,6 +161,22 @@ async function getUserPageProps(context: GetServerSidePropsContext) {
   });
 
   if (!eventData) {
+    const eventTypeByPreviousSlug = await prisma.eventType.findUnique({
+      where: {
+        userId_previousSlug: {
+          userId: user.id,
+          previousSlug: slug,
+        },
+      },
+    });
+    if (eventTypeByPreviousSlug) {
+      return {
+        redirect: {
+          destination: `/${username}/${eventTypeByPreviousSlug.slug}`,
+          permanent: true,
+        },
+      };
+    }
```

Comment:
Try to find the user `event-type` using `slug`, if not found use the `previousSlug` to find the `event-type` . If the `event-type` is found using the `previousSlug` it means we need to redirect to the `slug` of `eventTypeByPreviousSlug`